### ISSUE TITLE
useKeycloak hooks always returns keycloakStubInstance in next.js

### ISSUE THREAD

**Issue by rike422** [react-keycloak/react-keycloak#84]
**Describe the bug**

I am using @react-keycloak/ssr to build my next.js application
useKeycloak hook always returns a stub instance of keycloak so
createLoginUrl always returns an empty string.

I suspect that the problem is that the variable called initialized is always false, probably because of a problem with the SSR.
https://github.com/panz3r/react-keycloak/blob/master/packages/nextjs/src/useKeycloak.js#L11

I've tested it with react-keycloak-examples and I'm seeing similar symptoms.

I'm not sure if this is a bug or my lack of configuration.

**To Reproduce**

1. Get react-keycloak-examples
2. set keycloakCfg
3. yarn dev
4. Debug login button event

**Expected behavior**

I want to generate a link using createLoginUrl on the browser side.

**Desktop (please complete the following information):**

- OS: Ubuntu 
- Browser chrome
- Version 

**Additional context**

Next.js version 9.4.4

@panz3r 

Thank you for publishing a great library.
I gave a coffee fee through buymeacoffee.
This is my way of saying thank you.

**Comment by panz3r** [comment]
Hi @rike422 ,

Thanks for your feedback, unfortunately when running server-side the stub for Keycloak is required (because `keycloak-js` requires a `window` object to be defined which is not available on Node) and currently the implemented one does not provide access to all the methods of `keycloak-js`, instead of using the `createLoginUrl` you should consider binding the `login` method of the `keycloak` instance returned by `useKeycloak` to the button.

Currently I'm working on a newer version of `@react-keycloak/core` which will allow a better experience also when using keycloak on the server-side.

> Thank you for publishing a great library.
I gave a coffee fee through buymeacoffee.
This is my way of saying thank you.

Thanks, I really appreciate your support and I hope this library is helping you building your product as easily as possibile.

**Comment by rike422** [comment]
@panz3r 

Thank you for reply!

> unfortunately when running server-side the stub for Keycloak is required (because keycloak-js requires a window object to be defined which is not available on Node)

agree why stub is needed. though I have confirmed that stub is also used on the browser side.
I suspect that it is because the following line is not executed on the server side or context information is not carried over to the browser side.

https://github.com/panz3r/react-keycloak/blob/master/packages/nextjs/src/internals/context.js#L8

**Comment by rike422** [comment]
@panz3r 

Sorry. 
The problem was not solved.

I think the following line may not be executed in node.js or the state may not be inherited. When useKeycloak is executed in the browser, initialized is always false, so it's always returning a stub.
https://github.com/panz3r/react-keycloak/blob/master/packages/nextjs/src/internals/context.js#L8

I am using Next.js, but SSR is not required, so should I stop using SSRProvider and use KeycloakProvider?

I misunderstood that it was solved by the comment above because I forcibly set initialized to true in the forked react-keycloak project.

In that state, when using createLoginUrl, the [adapter](https://github.com/keycloak/keycloak/blob/master/adapters/oidc/js/src/main/resources/keycloak.js#L433) was undefined and did not work.
After changing createLoginUrl to login, the redirect was successful.

**Comment by rike422** [comment]
The following error is displayed when the javascript console is displaying a page.
This may be the cause because it is an error on the server side.
```
createReactKeycloakProvider
Uncaught ReferenceError: createReactKeycloakProvider is not defined at <anonymous>:1:1
``

**Comment by panz3r** [comment]
Hi @rike422 ,

That is a strange error, I'm going to look into this issue in the coming days.

Thanks again for your feedback.

**Comment by lbarthon** [comment]
Hi @panz3r,

We are having the same issue on next-js, if you could look into it it would be nice 😄 

Thanks for your help

**Comment by mattishii** [comment]
Also encountering this issue :(

**Comment by panz3r** [comment]
Hi @mattishii ,

The issue in the title should be resolved in the latest version (`v3.0.0`) of the `@react-keycloak/ssr`module.
